web/packages/teleport/src/Integrations/Enroll/AwsConsole/Access/Access.tsx

```diff
@@ -129,107 +190,145 @@ export function Access() {
+          Imported Profiles will be available as Resources with each Role
+          available as an account.
+        </P2>
+      </Box>
+      {!edit && status !== 'success' && intervalMs && (
+        <Alerts.Info details="It may take a moment for your integration to be ready before configuring access. This page will reload when ready.">
+          We&#39;re creating your integration
+        </Alerts.Info>
+      )}
+      {status === 'error' && !isNotFoundErr && (
         <Alerts.Danger details={error.message}>
           Error: {error.name}
         </Alerts.Danger>
       )}
-      {status === 'pending' && (
+      {/* only show the indicator for the first fetch */}
+      {status === 'pending' && !initialComplete && (
         <Box data-testid="loading" textAlign="center" m={10}>
           <Indicator />
         </Box>
       )}
-      {nothingToSync && <ProfilesEmptyState />}
-      {status === 'success' && data.profiles.length > 0 && (
+      {status === 'success' && !foundProfiles && <ProfilesEmptyState />}
+      {foundProfiles && (
         <Profiles
-          data={data.profiles}
-          fetchStatus={isFetching ? 'loading' : ''}
+          data={data?.profiles}
+          fetchStatus={isFetching || isRefetching ? 'loading' : ''}
           filters={filters}
           setFilters={setFilters}
           refetch={refetch}
           syncAll={syncAll}
           setSyncAll={setSyncAll}
+          isRefetching={isRefetching}
         />
       )}
-      {submitSync.error && (
-        <Alerts.Danger details={submitSync.error?.message} mt={2}>
-          Error: {submitSync.error.name}
+      {update.error && (
+        <Alerts.Danger details={update.error?.message} mt={2}>
+          Error: {update.error.name}
         </Alerts.Danger>
       )}
       <Flex gap={3} mt={3}>
         <ButtonPrimary
           width="200px"
-          onClick={() => submitSync.mutate()}
-          disabled={nothingToSync}
+          onClick={() => update.mutate()}
+          disabled={status !== 'success'}
         >
-          Enable Sync
+          {edit ? 'Update Sync' : 'Enable Sync'}
         </ButtonPrimary>
-        <ButtonSecondary
-          onClick={() =>
-            history.push(
-              cfg.getIntegrationEnrollRoute(
-                IntegrationKind.AWSRa,
-                'integration'
+        {edit ? (
+          <ButtonSecondary onClick={() => history.goBack()} width="100px">
+            Cancel
+          </ButtonSecondary>
+        ) : (
+          <ButtonSecondary
+            onClick={() =>
+              history.push(
+                cfg.getIntegrationEnrollRoute(
+                  IntegrationKind.AwsRa,
+                  'integration'
+                )
               )
-            )
-          }
-          width="100px"
-        >
-          Back
-        </ButtonSecondary>
+            }
+            width="100px"
+          >
+            Back
```

Comment:
this back button does take me back to the previous step, but the state wasn't saved which wasn't what i expected 😅